Allow the user to pre-create an empty Kubernetes secret for the unseal keys

**Is your feature request related to a problem? Please describe.**
In the use case when storing the Vault unseal keys to Kubernetes secret the Vault service-account currently requires the 'create' verb for secrets. This allows the pod to technically create any secret, which I would like to avoid.

**Describe the solution you'd like**
Allow the user to pre-create an _empty_ secret, which bank-vaults side car can populate.

**Describe alternatives you've considered**
Pre-create the secret with some initial dummy data, but this is not as clean as allowing an empty secret.

**Additional context**
Vault will not start up with empty secret, sidecar gives error:
```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/banzaicloud/bank-vaults/pkg/kv/k8s.(*k8sStorage).Set(0xc0007aad00, {0x20c6ba8, 0xa}, {0xc0003df000, 0xa, 0x10})
	/build/pkg/kv/k8s/k8s.go:102 +0x352
github.com/banzaicloud/bank-vaults/internal/vault.kvTester.Test({{0x7f341e5daf20?, 0xc0007aad00?}}, {0x20c6ba8, 0xa})
	/build/internal/vault/operator_client.go:124 +0x98
github.com/banzaicloud/bank-vaults/internal/vault.(*vault).Init(0xc0009083f0)
	/build/internal/vault/operator_client.go:268 +0xeb
main.glob..func4(0x3557a00?, {0x20bd764?, 0xc?, 0xc?})
	/build/cmd/bank-vaults/unseal.go:108 +0x716
github.com/spf13/cobra.(*Command).execute(0x3557a00, {0xc0007c00c0, 0xc, 0xc})
	/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x3557500)
	/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
main.execute()
	/build/cmd/bank-vaults/main.go:138 +0xd6
main.main()
	/build/cmd/bank-vaults/main.go:278 +0x5a
```
I can send a PR for this.

--- a/pkg/kv/k8s/k8s.go
+++ b/pkg/kv/k8s/k8s.go
@@ -99,6 +99,9 @@ func (k *k8sStorage) Set(key string, val []byte) error {
 		}
 		_, err = k.client.CoreV1().Secrets(k.namespace).Create(context.Background(), secret, metav1.CreateOptions{})
 	} else if err == nil {
+		if secret.Data == nil {
+			secret.Data = map[string][]byte{}
+		}
 		secret.Data[key] = val
 		_, err = k.client.CoreV1().Secrets(k.namespace).Update(context.Background(), secret, metav1.UpdateOptions{})
 	} else {

